### DISCUSSION TITLE
How does the extension work?

### DISCUSSION THREAD

**Discussion by KobeRiddle-dev** [aclap-dev/video-downloadhelper#817] [Help and Q&A]
### Your question

How does the extension work? Does it use a web service? Does it download the video through the extension itself (i.e. running natively on my machine)?

### Addon & CoApp version and details

Firefox. I haven't downloaded it yet.

**Comment by GrampaWildWilly** [comment]
I do not work for VDH.  I'm just another user.

There is a user manual of sorts you can read by clicking the Wiki link at the top of any page in this forum.  You would do well to read that.

The product is split into 2 parts.  There is the VDH browser extension that executes as part of the browser's address space.  Then there's the CoApp that executes independently in its own address space.  I gave an explanation of this in  this discussion: #744.  There is no external service or web site involved.  Everything executes on your computer.

As you read the Wiki, you will find beta versions of the browser extension.  For most software, beta versions are test balloons floated by the developer.  They are prone to not working.  This is not the case here.  Browser extensions need to go through a certain amount of red tape before they are accepted on the Mozilla/Google/Microsoft, respectively, web sites.  Beta versions are subject to somewhat less red tape.  Bug corrections go into the beta versions, which can appear fairly frequently, much sooner than they appear in the general release versions.  So you'd be doing yourself a favor by running a beta version.  You would skip the step of encountering a bug & having somebody here tell you, "Oh, that's been fixed in the beta."  I quit running public release versions of VDH at least 5 years ago.  Running the latest beta is my strong recommendation.

If you have more questions, do ask away.  Also, you would do well to read a few other discussions in this forum.  I know, there's a lot here & it can be overwhelming.  But you might read just the discussions listed on the first page of this forum.  That would be mostly discussions updated today.  It's a good idea to lurk in here for a while.  You'll learn a lot by just listening.
